src/rgw/rgw_data_sync.cc

```diff
@@ -4163,19 +4163,21 @@ class RGWBucketShardIsDoneCR: public RGWCoroutine {
         objv_tracker.clear();
         using ReadCR = RGWSimpleRadosReadCR<rgw_bucket_sync_status>;
         yield call(new ReadCR(sync_env->async_rados, sync_env->svc->sysobj,
-                              status_obj, &bucket_status, false, &objv_tracker));
+                              bucket_status_obj, &bucket_status, false, &objv_tracker));
         if (retcode < 0) {
           ldout(cct, 20) << "failed to read bucket shard status: "
               << cpp_strerror(retcode) << dendl;
           return set_cr_error(retcode);
         }
 
-        // exit with success to avoid stale shard being
-        // retried in error repo if we lost a race
         if (bucket_status.state != BucketSyncState::Incremental ||
             bucket_status.incremental_gen != generation) {
-            return set_cr_done();
-            }
+              // exit with success to avoid stale shard being
+              // retried in error repo if we lost a race
+              ldout(cct, 20) << "RGWBucketShardIsDoneCR: expected gen: " << generation <<
+              ", got: " << bucket_status.incremental_gen << dendl;
```

Comment:
this is for the `bucket_status.incremental_gen != generation` case. can you please split the `if (bucket_status.state != BucketSyncState::Incremental)` into a separate block with its own log message?

Reply:
sure, done.